### DISCUSSION TITLE
Namespaces Node missing

### DISCUSSION THREAD

**Discussion by johannwesely** [eclipse-milo/milo#1285] [General]
Hello, Our Customer asked for the Namespaces Server Node in OpcUaNamespace.java Line168 (in MiloDev1.0) is mentioned:
   ```
     // This Node is optional and we don't support it, so delete it entirely.
        serverTypeNode.getNamespacesNode().delete();
```
What can we do to support this Node ?

**Comment by kevinherron** [comment]
It looks like you might be able to create an instance of that Node (be sure to use the correct NodeId from namespace 0) and add it back into `OpcUaNamespace` with a reference to the Server Object.

I think there could also be a case made for simply not deleting it any more. I can't find any supporting evidence in the latest version of the spec that it shouldn't be there if no namespaces will be listed underneath it.

**Reply by johannwesely** [reply]
Hello Kevin, 
I tried to not delete the Namespaces Node and this seems for me to be the better solution as deleting it and adding it again (also i had Problems to archieve tthe adding correctly). 
And  it seems to work:

![Namespaces 2024-07-30 143951](https://github.com/user-attachments/assets/5cbf99aa-cb61-4bab-9ccc-1abe4d2c52db)

Is  it possible for you to remove the Deletion in OpcUaNamespace ?

**Reply by kevinherron** [reply]
I'll remove it.

I also have some WIP refactoring `CertificateValidator` so that `DefaultApplicationGroup` can be built with a custom validator (https://github.com/eclipse/milo/discussions/1281)
